prusti-tests/tests/verify/pass/time-resources/rec.rs

```diff
@@ -0,0 +1,25 @@
+// compile-flags: -Ptime_reasoning=true
+
+use prusti_contracts::*;
+
+#[extern_spec]
+impl usize {
+    #[pure]
+    #[ensures(self.pow(n) * self == self.pow(n + 1))]
```

Comment:
I'm surprised this isn't running into issues due to cycles. Can you also call `fib` from `main` with a concrete argument?

Reply:
By adding `#[ensures(self >= 1 ==> self.pow(0) == 1)]`, yes I can use it in main.